app/models/stat.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+class Stat
+  def initialize(name, show_in_ui: false, expose_via_api: false, &block)
+    @name = name
+    @show_in_ui = show_in_ui
+    @expose_via_api = expose_via_api
+    @block = block
+  end
+
+  attr_reader :name, :expose_via_api, :show_in_ui
+
+  def calculate
+    begin
+      return @block.call.transform_keys { |key| build_key(key) }
+    rescue StandardError => err
+      Discourse.warn_exception(
+        err,
+        message: "Unexpected error when collecting #{@name} About stats.",
+      )
+    end
+    {}
+  end
```

Comment:
It's a detail but when the full method is a begin/rescue, you can just have rescue.

Reply:
There is one more line after the rescue block, we want to return an empty hash in case something went wrong:

https://github.com/discourse/discourse/blob/721b38c99c77769f39c5f036577710f692931b09/app/models/stat.rb#L22

Reply:
why dont you return it from the rescue?

Reply:
Oh, that'll work, thank you!